web/src/components/navbar/NavBarDashboardLink.test.tsx

```diff
@@ -0,0 +1,95 @@
+import { NavBarDashboardLink } from "@/components/navbar/NavBarDashboardLink";
+import { ROUTES } from "@/lib/domain-logic/routes";
+import analytics from "@/lib/utils/analytics";
+import { mockPush, useMockRouter } from "@/test/mock/mockRouter";
+import {
+  WithStatefulUserData,
+  currentUserData,
+  setupStatefulUserDataContext,
+} from "@/test/mock/withStatefulUserData";
+import { Business } from "@businessnjgovnavigator/shared";
+import {
+  generateBusiness,
+  generateUser,
+  generateUserDataForBusiness,
+} from "@businessnjgovnavigator/shared/test";
+import { UserData, createEmptyUserData } from "@businessnjgovnavigator/shared/userData";
+import { fireEvent, render, screen, waitFor } from "@testing-library/react";
+
+jest.mock("next/router", () => ({ useRouter: jest.fn() }));
+jest.mock("@/lib/utils/analytics", () => setupMockAnalytics());
+jest.mock("@/lib/data-hooks/useUserData", () => ({ useUserData: jest.fn() }));
+jest.mock("@/lib/data-hooks/useRoadmap", () => ({ useRoadmap: jest.fn() }));
+
+function setupMockAnalytics(): typeof analytics {
+  return {
+    ...jest.requireActual("@/lib/utils/analytics").default,
+    event: {
+      ...jest.requireActual("@/lib/utils/analytics").default.event,
+      my_account: {
+        click: {
+          my_account: jest.fn(),
+        },
+      },
+    },
+  };
+}
+const mockAnalytics = analytics as jest.Mocked<typeof analytics>;
+
+const renderComponent = ({
+  userData,
+  previousBusiness,
+}: {
+  userData?: UserData | undefined;
+  previousBusiness?: Business | undefined;
+}): void => {
+  render(
+    <WithStatefulUserData
+      initialUserData={userData === undefined ? createEmptyUserData(generateUser({})) : userData}
+    >
+      <NavBarDashboardLink
+        businessName={"fake-business-name"}
+        linkText={"text-to-display"}
+        previousBusiness={previousBusiness}
+      />
+    </WithStatefulUserData>
+  );
+};
+
+describe("<NavBarDashboardLink/>", () => {
+  beforeEach(() => {
+    jest.resetAllMocks();
+    setupStatefulUserDataContext();
+  });
+
+  it("navigates to the dashboard when link is clicked by the user", async () => {
+    useMockRouter({});
+    renderComponent({});
+    fireEvent.click(screen.getByText("text-to-display"));
+    expect(mockPush).toHaveBeenCalledWith(ROUTES.dashboard);
+  });
+
+  it("fires the dashboard analytics event when link is clicked by the user", async () => {
+    useMockRouter({});
+    renderComponent({});
+    fireEvent.click(screen.getByText("text-to-display"));
+    expect(mockAnalytics.event.my_account.click.my_account).toHaveBeenCalled();
+  });
+
+  it("returns to dashboard and updates userData when link is clicked by the user and there is a previous business prop", async () => {
```

Comment:
dust - should the word `and` be replaced with `when`?

Reply:
I initially had it as `when` but changed it to and because it read as `when link is clicked by the user when there is a previous business prop` and it didn't sound well written.